src/cli/commands/force/lightning/local/device/create.ts

```diff
@@ -67,16 +85,26 @@ export class Create extends Setup {
     public deviceType: string = '';
 
     public async run(): Promise<any> {
-        await this.init(); // ensure init first
+        try {
```

Comment:
Incidentally, I saw your [previous comment](https://github.com/forcedotcom/lwc-dev-mobile-core/pull/35/files#r612030520) on this pattern. I don't know that there's a great way around it, since `SfdxCommand.init()` doesn't throw any more specific exceptions with its command validation. We could standardize the pattern, but that would involve (gasp!) creating a base `SfdxCommand`-inherited class for our plug-ins. Regardless of pros and cons of that approach, I'm in favor of a breather from noodling on inheritance chains at this point.

Reply:
The refactor exercise gave me an opportunity to learn several things:
* There aren't really lifecycle events with `SfdxCommand`. It is `run()` that start everything.
* We can leverage `validate` in `flagsConfig`. But we need to stay within the constraints of `SfdxCommand`, meaning, the method can be referenced from `static` and also it needs to be one of these types: `string | RegExp | ((val: string) => boolean)`
* `SfdxCommand` rightfully hides many of its internal types. My thinking is that creators of `SfdxCommand` want  developers like us to not go wild with it.

If we can play nicely with `SfdxCommand` I think it's a nice little framework that allows us to easily write commands. I agree with you, I definitely want to stay away from inherited class. That said there are more work that we can do around flags management&parameters validation. Those can be done in another round of refactoring in the future :)